src/python_testing/TC_OPCREDS_3_9.py

```diff
@@ -161,194 +168,490 @@ def get_unassigned_fabric_index(fabric_indices: list[int]) -> int:
+            asserts.assert_equal(exception_context.exception.status, Status.ConstraintError,
+                                 "Expected CONSTRAINT_ERROR for SetVIDVerificationStatement with VIDVerificationStatement too large")
+
+        with test_step(description="Establish a subscription to Operational Credentials cluster on endpoint 0 from TH1 fabric client, with MinIntervalFloor=0, MaxIntervalCeiling=30"):
+            attrib_listener = ClusterAttributeChangeAccumulator(opcreds)
+            await attrib_listener.start(cr1_dev_ctrl, cr1_dut_node_id, endpoint=0, min_interval_sec=0, max_interval_sec=30)
+
+        with test_step(description="Invoke SetVIDVerificationStatement with maximum-sized VVSC and VIDVerificationStatement present and setting VID to 0x6a01 on TH2's fabric, outside fail-safe. Verify VIDVerificationStatement, VVSC and VID updates are correct. Verify subscription received the updated values."):
+            vvsc = b"\xaa" * 400
+            VIDVerificationStatement = b"\x01" * VID_VERIFICATION_STATEMENT_SIZE_BYTES_V1
+            vendorID = 0x6a01
+
+            attrib_listener.reset()
+            await self.send_single_cmd(dev_ctrl=cr2_dev_ctrl, node_id=cr2_dut_node_id, cmd=opcreds.Commands.SetVIDVerificationStatement(VIDVerificationStatement=VIDVerificationStatement, vvsc=vvsc, vendorID=vendorID))
+            attrib_listener.await_all_expected_report_matches(expected_matchers=[make_vid_matcher(
+                cr2_fabric_index, vendorID), make_vvs_matcher(cr2_fabric_index, VIDVerificationStatement)], timeout_sec=30.0)
+
+            updated_fabrics, updated_nocs = await self.read_updated_fabrics(dev_ctrl=cr1_dev_ctrl, node_id=cr1_dut_node_id)
+
+            cr2_fabrics_entry = get_entry_for_fabric(cr2_fabric_index, updated_fabrics)
+            asserts.assert_is_not_none(
+                cr2_fabrics_entry, f"Could not find Fabrics list entry for TH2's fabric index {cr2_fabric_index}")
+
+            cr2_nocs_entry = get_entry_for_fabric(cr2_fabric_index, updated_nocs)
+            asserts.assert_is_not_none(cr2_nocs_entry, f"Could not find NOCs list entry for TH2's fabric index {cr2_fabric_index}")
+
+            asserts.assert_equal(cr2_nocs_entry.vvsc, vvsc,
+                                 "Did not get the expected value set for VVSC field of NOCs list for TH2's fabric.")
+            asserts.assert_equal(cr2_fabrics_entry.VIDVerificationStatement, VIDVerificationStatement,
+                                 "Did not get the expected value set for VIDVerificationStatement field of Fabrics list for TH2's fabric.")
+            asserts.assert_equal(cr2_fabrics_entry.vendorID, vendorID,
+                                 "Did not get the expected value set for VendorID field of Fabrics list for TH2's fabric.")
+
+        with test_step(description="Invoke SetVIDVerificationStatement with maximum-sized VVSC on TH1's fabric, outside fail-safe. Verify INVALID_COMMAND due to presence of ICAC."):
```

Comment:
As far as I can tell, the command description doesn't require a failsafe, does it? Does the behaviour change when the failsafe is armed if there wasn't a prior add/update noc call?